dspace-server-webapp/src/main/java/org/dspace/app/rest/MetadataFieldNameRestController.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.sql.SQLException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.dspace.app.rest.converter.ConverterService;
+import org.dspace.app.rest.model.MetadataFieldRest;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.MetadataField;
+import org.dspace.content.service.MetadataFieldService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller will handle all the incoming calls on the /api/core/metadatafields/name/<:metadata-field-full-name>
+ * endpoint where the metadata-field-full-name parameter can be filled in to match a specific metadata field by name
+ * There's always at most one metadata field per name.
+ * <p>
+ * It responds with:
+ * <p>
+ * The single metadata field if there's a match
+ * 404 if the metadata field doesn't exist
+ *
+ * @author Maria Verdonck (Atmire) on 17/07/2020
+ */
+@RestController
+@RequestMapping("/api/" + MetadataFieldRest.CATEGORY + "/" + MetadataFieldRest.NAME_PLURAL)
+public class MetadataFieldNameRestController {
```

Comment:
This Controller isn't needed as we don't have a `/api/core/metdatafields/name/[:metadata-field-full-name]` path in the Contract: https://github.com/DSpace/Rest7Contract/blob/main/metadatafields.md

Instead, this endpoint is at `/api/core/metadatafields/search/byFieldName` and you've already implemented that endpoint in the `MetadataFieldRestRepository` in this same PR.  So, this entire Controller can be removed.

Reply:
@tdonohue **A quick overview for the 'need' of this MetadataFieldNameRestController class**:

For the front end validation on whether or not a filled in mdField is valid (on the edit item > metadata page) an exact search of mdField is needed, where either exactly one matching mdField is given or none if it doesn’t exist. While this could be done by the findByName endpoint, this would require:
* For mdFields without a qualifier > Index a null/unknown value in solr
* In frontend add &qualifier=null along to the restend when searching for an mdField without a qualifier
* Example: You want to validate whether or not dc.title that the user has filled in is a valid mdField
    * So you use searchByName?schema=dc&element=title which returns multiple results (dc.title, dc.title.other, dc.title.en, …). 
    * But if for example dc.title does not exist, this search would still return multiple (other) results.

Another solution might be to go over these results in the frontend and look for an exact match, but is not a frontend responsibility.
This same endpoint however is used to give the list of suggestions while the user is typing, so it would be hard to know when to add that &qualifier=null, because the user could just still be typing, and has yet to add the qualifier.

Either way for a single exact match a search in general is not recommended, hence the new /api/core/metadatafields/name/<full-md-name> endpoint. Which doesn’t return an array or embed, but a single field or a 404 if no matching field is found, which uses the DB and not the Solr indexed mdFields.
Previous implementation of suggestion/validation on the edit item page disabled this here: https://github.com/DSpace/dspace-angular/pull/749/commits/09f1b7e01fac6746c373e0497ac9971800ffc8a7

We are currently in the process of creating the validator related to https://github.com/DSpace/dspace-angular/issues/750 and that was the reason that this has come up

Reply:
@tdonohue As per described in https://github.com/DSpace/Rest7Contract/pull/145, 
The code has now been altered to remove this search functionality for a single field, with the complete name, in the separate MetadataFieldNameRestController, and it has moved to the more general search in MetadataFieldRestRepository

The unrelated tests are still failing here because of the MetadataField being indexing on add/edit/deletes now.
This will need to be looked into further to make sure the solr changes don't have any effect on other tests.